src/domain_extended/mod.rs

```diff
@@ -0,0 +1,205 @@
+//!
+//! A module for extending the domain of an arbitrary homomorphic commitment scheme beyond the 
+//! maximum degree supported by it. 
+
+mod data_structures;
+pub use data_structures::*;
+
+use crate::{Error, LinearCombination, Polynomial, PolynomialCommitment};
+use crate::{PCProof, PCCommitterKey};
+use algebra::{
+    groups::{Group, GroupVec},
+    fields::Field,
+    curves::Curve,
+};
+use std::marker::PhantomData;
+use rand_core::RngCore;
+
+/// The domain extension of a given homomorphic commitment scheme `PC`.
+#[derive(Derivative)]
+#[derivative(Clone(bound = ""))]
+pub struct DomainExtendedPolynomialCommitment<G: Curve, PC: PolynomialCommitment<G, Commitment = G>> {
```

Comment:
I don't really like this being a struct, probably a trait extending PolynomialCommitment (to be implemented too by IPA_PC) would've been better, also considering the way we will use it (I think we are always going to use the DomainExtended variant) and its generality. 
We can define in this trait only the function and types that we expect to change with respect to PolynomialCommitment trait, e.g.:

types: Commitment, Randomness, MultiPointProof
functions: commit, open, succinct_verify

And give them default implementations.
We give a blanket implementation of this trait for IPA.
These should work too without the need to disambiguate (a function will be templatized either by PolynomialCommitment or DomainExtendedPolynomialCommitment).
At worst, we need to disambiguate the calls to the common functions.
<Type as PolynomialCommitment>::commit() or <Type as DomainExtendedPolynomialCommitment>::commit().

What do you think ?

Reply:
In terms of traits, I do not see anything that discriminates a domain extended scheme from a non-extended scheme.